plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/connection/QLspConnectionProvider.java

```diff
@@ -78,6 +80,28 @@ protected final void addEnvironmentVariables(final Map<String, String> env) {
         }
     }
 
+    private String getCaCert() {
+        String caCertPreference = Activator.getDefault().getPreferenceStore().getString(AmazonQPreferencePage.CA_CERT);
+        if (!StringUtils.isEmpty(caCertPreference)) {
+            Activator.getLogger().info("Using user-defined CA cert: " + caCertPreference);
+            return caCertPreference;
+        }
+
+        try {
+            String pemContent = ProxyUtil.getCertificatesAsPem();
```

Comment:
this is probably fine, but since eclipse does not really have dynamic certificate mutation it is probably safe to make one copy for the IDE session